asl_xdsl/dialects/asl.py

```diff
@@ -396,6 +457,25 @@ def __init__(self, arg: SSAValue, attr_dict: Mapping[str, Attribute] = {}):
         )
 
 
+@irdl_op_definition
+class BoolToI1Op(IRDLOperation):
+    """A hack to convert !asl.bool to i1 so that we can use scf.if."""
+
+    name = "asl.bool_to_i1"
+
+    arg = operand_def(BoolType())
+    res = result_def(builtin.IntegerType(1))
+
+    assembly_format = "$arg `:` type($arg) `->` type($res) attr-dict"
```

Comment:
I'm not sure why, but the attr dict tends to be before the type by convention

```suggestion
    assembly_format = "$arg attr-dict `:` type($arg) `->` type($res)"
```

Reply:
Actually, that's because I moved it to the end on the other operations.
Mostly because I hate being it at the beginning, because it is "optional" in a way.

Reply:
Are there other dialects that do this? Seems to me like we should stick with the convention if not.

Reply:
(Not changed - unsure which to go with)